test-case-creation-strategy.mdx

```diff
@@ -1,195 +1,167 @@
+  <img
+    src="images/bestpractice/dependency-01-accept-cookies.png"
+    alt="no dependency for accept cookies"
+    width="300px"
+  />
+</Frame>
 
-You can easily delete unnecessary steps by clicking the trash can symbol.
+<div class="mt-8" />
 
-<Frame caption="delete obsolete steps, screenshot 02/2024">
+<Frame caption="'login' test depends on 'accept cookies', 08/2024">
   <img
-    src="images/steps-delete-step.png"
-    alt="delete obsolete steps"
+    src="images/bestpractice/dependency-02-login.png"
+    alt="login test depends on accept cookies test"
     width="300px"
   />
 </Frame>
 
-### Run test cases
-
-Running a test case is producing a trace. A trace is a detailed description of what happened during test case execution. Use the trace for debugging.
+<div class="mt-8" />
 
-<Frame caption="run test case for validation, screenshot 02/2024">
+<Frame caption="'navigate to returns' test depends on 'login' test, 08/2024">
   <img
-    src="images/run-test-case.png"
-    alt="run test case for validation"
+    src="images/bestpractice/dependency-03-navigate-returns.png"
+    alt="Navigate to returns test depends on login test"
     width="300px"
   />
 </Frame>
 
+<div class="mt-8" />
+
+<Tip>
+**Advantages of using dependencies**
+
+- code is reused and only maintained in one place
+- much shorter test cases are easier to understand and debug
+- better chance of the AI agent being successful when generating test steps, it can focus on the main task
+
+</Tip>
+
+## Help the AI agent when it went wrong
+
+We are constantly improving the AI agent but it still might make mistakes. Help it find its way. Use the `steps` tab to edit and fine tune your test.
+
+You can add missing steps by clicking the `+ add step` button and choosing the desired **interaction** or **assertion**. Use the **virtual locator picker** to select the right `locator` in the snapshot on the right. Drag and drop the step where needed.
+
+More details can be found in the [editing test cases](/edit-test-case) section.
+
+<Frame caption="add missing steps, 08/2024">
+  <img src="images/bestpractice/add-steps.png" alt="add missing steps" />
+</Frame>
+
+<div class="mt-8" />
+
 <Tip>
   
-  **Advantages of helping out the agent**
+  **Advantages of manual editing**
 
-- modifying steps manually is typically much faster than changing the prompts
-- You have much more control over the outcome.
+- Modifying steps much faster than changing the prompts and having the step AI auto-generated
+- Our virtual locator picker, copy paste and drag & drop functionalities make it a smooth experience
+- You have much more control over the outcome
 
 </Tip>
 
-## Use `run locally`
+## Run Octomind tests locally
 
-`run locally` helps you to save execution time and you can directly see what is happening.
+`Run locally` is most helpful for step by step adjustments since it is giving you the current state of a page. It will noticeably speed-up your test creation efforts.
```

Comment:
speed up

Reply:
```suggestion
`Run locally` is most helpful for step by step adjustments since it is giving you the current state of a page. It will noticeably speed up your test creation efforts.
```